cmd/gh/main.go

```diff
@@ -91,10 +94,57 @@ func main() {
 		}
 	}
 
+	// TODO support other names
+	ghtoken := os.Getenv("GITHUB_TOKEN")
+	if ghtoken != "" {
+		skipAuthCheck = true
+	}
+
+	if !skipAuthCheck {
+		hasAuth := false
+
+		cfg, err := cmdFactory.Config()
+		if err == nil {
+			hasAuth = cmdutil.CheckAuth(cfg)
+		}
+
+		if !hasAuth {
+			authMessage := heredoc.Doc(`
+       %s                   'cxO0KK0Oxc'
+                                            'oONMMMMMMMMMMMMNOo'
+                                         'dXMMMMMMMMMMMMMMMMMMMMXd'
+                                       .kMMMMMMMMMMMMMMMMMMMMMMMMMMk.
+       To authenticate, please        lWMMMM:.'ckXKOkkkkOKXkc'.:MMMMWl
+       run 'gh auth login'.          oMMMMMM.                  .MMMMMMo
+                                    ;WMMMMMX.                  .XMMMMMW;
+                                    0MMMMMK.                    .KMMMMM0
+       You can also set the         MMMMMMl                      lMMMMMM
+       GITHUB_TOKEN environment     MMMMMMd                      dMMMMMM
+       variable, if preferred.      KMMMMMN'                    'NMMMMMK
+                                    cMMMMMMNc                  cNMMMMMMc
+                                     xMMNdkXMNkl;,.      .,;lkNMMMMMMMx
+                                      xWMNx;kWMMMK.      .KMMMMMMMMMWx
+                                       ,KMM0;.;:;.        dMMMMMMMMK,
+                                         :OWMXOOOc        dMMMMMWO:
+                                           .:kXMMd        dMMXk:.
```

Comment:
ASCII art 😻

Reply:
I put in a `TerminalSize` check so as not to print this if the terminal width is too narrow (thanks @tierninho for pointing that out)